### PR TITLE
feature: prevent page navigation on horizontal scroll

### PR DESCRIPTION
**Pull request by LiKang6688**
prevent the page going back and next to visited pages on a horizontal scroll

### CODE REVIEW

**Comment by haxxmaxx** [comment]
So do you have the same settings for changing page and scrolling horizontally? I don't and it works fine for me

**Comment by LiKang6688** [comment]
> So do you have the same settings for changing page and scrolling horizontally? I don't and it works fine for me

the same settings. what do you mean?

**Comment by haxxmaxx** [comment]
> > So do you have the same settings for changing page and scrolling horizontally? I don't and it works fine for me
> 
> the same settings. what do you mean?

On your mac. With the trackpad, I use two fingers to scroll and three to go back/forward

**Comment by LiKang6688** [comment]
> > > So do you have the same settings for changing page and scrolling horizontally? I don't and it works fine for me
> > 
> > 
> > the same settings. what do you mean?
> 
> On your mac. With the trackpad, I use two fingers to scroll and three to go back/forward. I think the default settings might overlap

Yes, in the default setting, you can use two fingers to scroll and go back/forward.
Screenshot 2021-05-05 at 12 05 43

**Comment by haxxmaxx** [comment]
> > > > So do you have the same settings for changing page and scrolling horizontally? I don't and it works fine for me
> > > 
> > > 
> > > the same settings. what do you mean?
> > 
> > 
> > On your mac. With the trackpad, I use two fingers to scroll and three to go back/forward. I think the default settings might overlap
> 
> Yes, in the default setting, you can use two fingers to scroll and go back/forward.
> Screenshot 2021-05-05 at 12 05 43

Yes, so I don't think we should compensate for that.

**Comment by LiKang6688** [comment]
> > > > > So do you have the same settings for changing page and scrolling horizontally? I don't and it works fine for me
> > > > 
> > > > 
> > > > the same settings. what do you mean?
> > > 
> > > 
> > > On your mac. With the trackpad, I use two fingers to scroll and three to go back/forward. I think the default settings might overlap
> > 
> > 
> > Yes, in the default setting, you can use two fingers to scroll and go back/forward.
> > Screenshot 2021-05-05 at 12 05 43
> 
> Yes, so I don't think we should compensate for that.
you do not think the PR is needed? from my previous experience, it is a wanted feature from end-users.

File: src/table/handle-scroll.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const handleScroll = (evt, tableSection) => {
+  evt.stopPropagation();
+  const max = tableSection.current.scrollWidth - tableSection.current.offsetWidth;
+  if (
+    max > 0 &&
+    (tableSection.current.scrollLeft + evt.deltaX < 0 || tableSection.current.scrollLeft + evt.deltaX > max)
```
Comment (haxxmaxx) [inline src/table/handle-scroll.js:6]: `tableSection.current.scrollLeft` is used 3 times, save it as a constant

Comment (LiKang6688) [reply src/table/handle-scroll.js:6]: sure

File: src/table/handle-scroll.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const handleScroll = (evt, tableSection) => {
+  evt.stopPropagation();
+  let { scrollLeft } = tableSection.current;
+  const { deltaX } = evt;
+  const max = tableSection.current.scrollWidth - tableSection.current.offsetWidth;
+  if (max > 0 && (scrollLeft + deltaX < 0 || scrollLeft + deltaX > max)) {
+    evt.preventDefault();
+    scrollLeft = Math.max(0, Math.min(max, scrollLeft + deltaX));
```
Comment (haxxmaxx) [inline src/table/handle-scroll.js:8]: I meant like this
```suggestion
  const newScrollLeft = tableSection.current.scrollLeft + evt.deltaX;
  const max = tableSection.current.scrollWidth - tableSection.current.offsetWidth;
  if (max > 0 && (newScrollLeft < 0 || newScrollLeft > max)) {
    evt.preventDefault();
    scrollLeft = Math.max(0, Math.min(max, newScrollLeft));
```

Comment (LiKang6688) [reply src/table/handle-scroll.js:8]: scrollLeft in line 7 does not seem correct

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**